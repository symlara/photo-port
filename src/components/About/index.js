import React from 'react';
import coverImage from "../../assets/cover/cover-image.jpg";
// about component
function About() {
    return (
        <section className="my-5">
            <h1 id="about">Who am I?</h1>
            <img src={coverImage} className="my-2" style={{ width: "100%" }} alt="cover" />
            <div className="my-2">
                <p>Anim laboris ad tempor ullamco exercitation consequat commodo sit reprehenderit elit in occaecat mollit non. Elit nisi adipisicing consectetur officia eiusmod sint minim Lorem. Culpa fugiat irure magna tempor excepteur consequat anim et fugiat dolor occaecat quis officia sit. Ipsum sit voluptate velit et elit non.</p>
            </div>
        </section>
        
    );
}

export default About;