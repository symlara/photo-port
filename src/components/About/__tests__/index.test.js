//We need react to enable the components to function properly so we can test them.
import React from 'react';
// "render" the component.
import { render, cleanup } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
//we'll import the component we will be testing,
import About from '..';
//This will ensure that after each test, we won't have any leftover memory data that could give us false results.
afterEach(cleanup)

describe('About component', () => {
    // renders About test
    //first test
    it('renders', () => {
        render(<About />);
    });

    // second test
    it('matches snapshot DOM node structure', () => {
        // render About
        const  { asFragment } = render(<About />);
        expect(asFragment()).toMatchSnapshot();
    })
})