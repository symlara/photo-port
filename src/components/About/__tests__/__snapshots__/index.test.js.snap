// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1
      id="about"
    >
      Who am I?
    </h1>
    <img
      alt="cover"
      class="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
    <div
      class="my-2"
    >
      <p>
        Anim laboris ad tempor ullamco exercitation consequat commodo sit reprehenderit elit in occaecat mollit non. Elit nisi adipisicing consectetur officia eiusmod sint minim Lorem. Culpa fugiat irure magna tempor excepteur consequat anim et fugiat dolor occaecat quis officia sit. Ipsum sit voluptate velit et elit non.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
